apiserver/facades/client/client/status.go

```diff
@@ -621,19 +639,30 @@ func fetchAllApplicationsAndUnits(
 		allBindingsByApp[bindings.AppName] = bindings.Bindings
 	}
 
+	charmProfileNames := make(map[charm.URL]lxdProfileName)
 	for _, app := range applications {
 		appMap[app.Name()] = app
 		appUnits := allUnitsByApp[app.Name()]
+		charmURL, _ := app.CharmURL()
```

Comment:
you should be able to get the profile from app.Charm().LXDProfile() also?  Not sure if that simplifies things or not?

Reply:
Potentially.